MW errors on PHP 5.5 and 5.6

Build:
* CentOS 6.6
* Apache 2.4
* MySQL 5.6.25
* PHP 5.5.26 or PHP 5.6.10 (both produce this issue)

MediaWiki seems to install fine, but upon trying to log in the following error is displayed:
>Fatal error: Maximum execution time of 30 seconds exceeded in /var/www/meza1/htdocs/wiki/includes/utils/MWCryptRand.php on line 317

I just ran an install with PHP v5.5.29 on CentOS 6.6. Unfortunately this problem did not magically disappear. I ran it on the `centos7` branch because it loads lib_mcrypt from EPEL instead of RPMForge (due to changes in CentOS 7). I was hoping this would have an effect.

This may have some effect on #16 (enable object caching) since OpCache is compiled by default with PHP 5.5+. It can be added to PHP 5.2+, but that would require different installations for PHP 5.4 versus 5.5 and 5.6.

Last we tried installing newer PHP versions we didn't have the error logs. Here's the relevant PHP errors:

```
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c: In function ‘php_openssl_setup_crypto’:
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:454: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:462: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:471: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:476: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:480: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:488: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:497: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/openssl/xp_ssl.c:502: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/ext/zlib/zlib.c:1035: warning: initialization from incompatible pointer type
/root/sources/php-5.5.29/ext/curl/interface.c:74:5: warning: #warning "libcurl was compiled with SSL support, but configure could not determine which" "library was used; thus no SSL crypto locking callbacks will be set, which may " "cause random crashes on SSL requests"
/root/sources/php-5.5.29/ext/curl/interface.c: In function ‘zif_curl_getinfo’:
/root/sources/php-5.5.29/ext/curl/interface.c:3124: warning: call to ‘_curl_easy_getinfo_err_curl_slist’ declared with attribute warning: curl_easy_getinfo expects a pointer to struct curl_slist * for this info
/root/sources/php-5.5.29/ext/curl/interface.c:3159: warning: call to ‘_curl_easy_getinfo_err_curl_slist’ declared with attribute warning: curl_easy_getinfo expects a pointer to struct curl_slist * for this info
/root/sources/php-5.5.29/ext/curl/interface.c: In function ‘_php_curl_setopt’:
/root/sources/php-5.5.29/ext/curl/interface.c:2738: warning: call to ‘_curl_easy_setopt_err_progress_cb’ declared with attribute warning: curl_easy_setopt expects a curl_progress_callback argument for this option
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp: In function ‘void zif_breakiter_set_text(int, zval*, zval**, zval*, int, void***)’:
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp:165: warning: ‘int zend_get_parameters_ex(int, ...)’ is deprecated (declared at /root/sources/php-5.5.29/Zend/zend_API.h:238)
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp:165: warning: ‘int zend_get_parameters_ex(int, ...)’ is deprecated (declared at /root/sources/php-5.5.29/Zend/zend_API.h:238)
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp: In function ‘void zif_breakiter_next(int, zval*, zval**, zval*, int, void***)’:
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp:278: warning: ‘int zend_get_parameters_ex(int, ...)’ is deprecated (declared at /root/sources/php-5.5.29/Zend/zend_API.h:238)
/root/sources/php-5.5.29/ext/intl/breakiterator/breakiterator_methods.cpp:278: warning: ‘int zend_get_parameters_ex(int, ...)’ is deprecated (declared at /root/sources/php-5.5.29/Zend/zend_API.h:238)
/root/sources/php-5.5.29/ext/mbstring/libmbfl/filters/mbfilter_iso2022jp_mobile.c:51: warning: initialization from incompatible pointer type
/root/sources/php-5.5.29/ext/mcrypt/mcrypt_filter.c: In function ‘php_mcrypt_filter_create’:
/root/sources/php-5.5.29/ext/mcrypt/mcrypt_filter.c:210: warning: passing argument 1 of ‘mcrypt_module_open’ discards qualifiers from pointer target type
/usr/include/mutils/mcrypt.h:38: note: expected ‘char *’ but argument is of type ‘const char *’
/root/sources/php-5.5.29/ext/session/session.c: In function ‘ps_srlzr_decode_php_serialize’:
/root/sources/php-5.5.29/ext/session/session.c:866: warning: passing argument 2 of ‘php_var_unserialize’ from incompatible pointer type
/root/sources/php-5.5.29/ext/standard/php_var.h:53: note: expected ‘const unsigned char **’ but argument is of type ‘const char **’
/root/sources/php-5.5.29/ext/standard/browscap.c: In function ‘zm_startup_browscap’:
/root/sources/php-5.5.29/ext/standard/browscap.c:339: warning: passing argument 3 of ‘ts_allocate_id’ from incompatible pointer type
/root/sources/php-5.5.29/Zend/../TSRM/TSRM.h:111: note: expected ‘ts_allocate_ctor’ but argument is of type ‘void (*)(struct zend_browscap_globals *, void ***)’
/root/sources/php-5.5.29/ext/standard/exec.c: In function ‘php_escape_shell_cmd’:
/root/sources/php-5.5.29/ext/standard/exec.c:257: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/exec.c:257: warning: initialization makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/exec.c: In function ‘php_escape_shell_arg’:
/root/sources/php-5.5.29/ext/standard/exec.c:354: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/exec.c:354: warning: initialization makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/file.c: In function ‘php_fgetcsv_lookup_trailing_spaces’:
/root/sources/php-5.5.29/ext/standard/file.c:1776: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:1776: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:1776: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/file.c:1781: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:1781: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c: In function ‘php_fgetcsv’:
/root/sources/php-5.5.29/ext/standard/file.c:2049: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2049: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2075: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2075: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2075: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/file.c:2164: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2164: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2219: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2219: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2219: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/file.c:2232: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2232: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2243: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2243: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2243: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/file.c:2263: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2263: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2274: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2274: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/file.c:2274: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/string.c: In function ‘php_basename’:
/root/sources/php-5.5.29/ext/standard/string.c:1435: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/string.c:1435: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/string.c:1435: warning: assignment makes integer from pointer without a cast
/root/sources/php-5.5.29/ext/standard/string.c:1441: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/ext/standard/string.c:1441: warning: pointer/integer type mismatch in conditional expression
/root/sources/php-5.5.29/Zend/zend_compile.c: In function ‘zend_traits_copy_functions’:
/root/sources/php-5.5.29/Zend/zend_compile.c:4021: warning: assignment discards qualifiers from pointer target type
/root/sources/php-5.5.29/sapi/fpm/fpm/fpm_main.c:893: warning: initialization from incompatible pointer type
PEAR package PHP_Archive not installed: generated phar will require PHP's phar extension be enabled.
/usr/local/apache2/build/instdso.sh SH_LIBTOOL='/usr/local/apache2/build/libtool' libphp5.la /usr/local/apache2/modules
libtool: install: warning: remember to run `libtool --finish /root/sources/php-5.5.29/libs'
```

--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -56,11 +56,12 @@ read usergithubtoken
 usergithubtoken=${usergithubtoken:-$default_usergithubtoken}
 
 # Prompt user for PHP version
-default_phpversion="5.4.42"
-echo -e "\nVisit http://php.net/downloads.php for version numbers"
-echo -e "Type the version of PHP you would like (such as 5.4.42) and press [ENTER]:"
-read -e -i $default_phpversion phpversion
-phpversion=${phpversion:-$default_phpversion}
+default_phpversion="5.6.14"
+phpversion=$default_phpversion #hard code version for now based on #24
+# echo -e "\nVisit http://php.net/downloads.php for version numbers"
+# echo -e "Type the version of PHP you would like (such as 5.4.42) and press [ENTER]:"
+# read -e -i $default_phpversion phpversion
+# phpversion=${phpversion:-$default_phpversion}
 
 # Prompt user for MySQL password
 default_mysql_root_pass=`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`

--- a/scripts/php.ini-development
+++ b/scripts/php.ini-development
@@ -742,7 +742,7 @@ enable_dl = Off
 ; will look for to know it is OK to continue execution.  Setting this variable MAY
 ; cause security issues, KNOW WHAT YOU ARE DOING FIRST.
 ; http://php.net/cgi.redirect-status-env
-;cgi.redirect_status_env = 
+;cgi.redirect_status_env =
 
 ; cgi.fix_pathinfo provides *real* PATH_INFO/PATH_TRANSLATED support for CGI.  PHP's
 ; previous behaviour was to set PATH_TRANSLATED to SCRIPT_FILENAME, and to not grok
@@ -883,8 +883,8 @@ default_socket_timeout = 60
 ;extension=php_pspell.dll
 ;extension=php_shmop.dll
 
-; The MIBS data available in the PHP distribution must be installed. 
-; See http://www.php.net/manual/en/snmp.installation.php 
+; The MIBS data available in the PHP distribution must be installed.
+; See http://www.php.net/manual/en/snmp.installation.php
 ;extension=php_snmp.dll
 
 ;extension=php_soap.dll
@@ -1829,3 +1829,104 @@ ldap.max_links = -1
 ; Local Variables:
 ; tab-width: 4
 ; End:
+
+[opcache]
+; add OPcache file location
+zend_extension=/usr/local/php/lib/php/extensions/no-debug-zts-20131226/opcache.so
+
+; Determines if Zend OPCache is enabled
+opcache.enable=1
+
+; Determines if Zend OPCache is enabled for the CLI version of PHP
+opcache.enable_cli=1
+
+; The OPcache shared memory storage size
+opcache.memory_consumption=256
+
+; The amount of memory for interned strings in Mbytes
+opcache.interned_strings_buffer=4
+
+; The maximum number of keys (scripts) in the OPcache hash table
+; Only numbers between 200 and 100000 are allowed
+opcache.max_accelerated_files=20000
+
+; The maximum percentage of "wasted" memory until a restart is scheduled
+;opcache.max_wasted_percentage=5
+
+; When this directive is enabled, the OPcache appends the current working
+; directory to the script key, thus eliminating possible collisions between
+; files with the same name (basename). Disabling the directive improves
+; performance, but may break existing applications.
+;opcache.use_cwd=1
+
+; When disabled, you must reset the OPcache manually or restart the
+; webserver for changes to the filesystem to take effect
+opcache.validate_timestamps=1
+
+; How often (in seconds) to check file timestamps for changes to the shared
+; memory storage allocation. ("1" means validate once per second, but only
+; once per request. "0" means always validate)
+opcache.revalidate_freq=30
+
+; Enables or disables file search in include_path optimization
+;opcache.revalidate_path=0
+
+; If disabled, all PHPDoc comments are dropped from the code to reduce the
+; size of the optimized code.
+;opcache.save_comments=1
+
+; If disabled, PHPDoc comments are not loaded from SHM, so "Doc Comments"
+; may be always stored (save_comments=1), but not loaded by applications
+; that don't need them anyway.
+;opcache.load_comments=1
+
+; If enabled, a fast shutdown sequence is used for the accelerated code
+;opcache.fast_shutdown=0
+
+; Allow file existence override (file_exists, etc.) performance feature.
+;opcache.enable_file_override=0
+
+; A bitmask, where each bit enables or disables the appropriate OPcache
+; passes
+;opcache.optimization_level=0xffffffff
+
+;opcache.inherited_hack=1
+;opcache.dups_fix=0
+
+; The location of the OPcache blacklist file (wildcards allowed).
+; Each OPcache blacklist file is a text file that holds the names of files
+; that should not be accelerated. The file format is to add each filename
+; to a new line. The filename may be a full path or just a file prefix
+; (i.e., /var/www/x  blacklists all the files and directories in /var/www
+; that start with 'x'). Line starting with a ; are ignored (comments).
+;opcache.blacklist_filename=
+
+; Allows exclusion of large files from being cached. By default all files
+; are cached.
+;opcache.max_file_size=0
+
+; Check the cache checksum each N requests.
+; The default value of "0" means that the checks are disabled.
+;opcache.consistency_checks=0
+
+; How long to wait (in seconds) for a scheduled restart to begin if the cache
+; is not being accessed.
+; Default was 180 (3 minutes), changed to 1 hour to keep the
+; cache from being reset regularly.
+opcache.force_restart_timeout=3600
+
+; OPcache error_log file name. Empty string assumes "stderr".
+; opcache.error_log=/opt/meza/logs/opcache_error.log
+
+; All OPcache errors go to the Web server log.
+; By default, only fatal errors (level 0) or errors (level 1) are logged.
+; You can also enable warnings (level 2), info messages (level 3) or
+; debug messages (level 4).
+;opcache.log_verbosity_level=1
+
+; Preferred Shared Memory back-end. Leave empty and let the system decide.
+;opcache.preferred_memory_model=
+
+; Protect the shared memory from unexpected writing during script execution.
+; Useful for internal debugging only.
+;opcache.protect_memory=0

--- a/scripts/php.ini-production
+++ b/scripts/php.ini-production
@@ -883,8 +883,8 @@ default_socket_timeout = 60
 ;extension=php_pspell.dll
 ;extension=php_shmop.dll
 
-; The MIBS data available in the PHP distribution must be installed. 
-; See http://www.php.net/manual/en/snmp.installation.php 
+; The MIBS data available in the PHP distribution must be installed.
+; See http://www.php.net/manual/en/snmp.installation.php
 ;extension=php_snmp.dll
 
 ;extension=php_soap.dll
@@ -1829,3 +1829,104 @@ ldap.max_links = -1
 ; Local Variables:
 ; tab-width: 4
 ; End:
+
+[opcache]
+; add OPcache file location
+zend_extension=/usr/local/php/lib/php/extensions/no-debug-zts-20131226/opcache.so
+
+; Determines if Zend OPCache is enabled
+opcache.enable=1
+
+; Determines if Zend OPCache is enabled for the CLI version of PHP
+opcache.enable_cli=1
+
+; The OPcache shared memory storage size
+opcache.memory_consumption=256
+
+; The amount of memory for interned strings in Mbytes
+opcache.interned_strings_buffer=4
+
+; The maximum number of keys (scripts) in the OPcache hash table
+; Only numbers between 200 and 100000 are allowed
+opcache.max_accelerated_files=20000
+
+; The maximum percentage of "wasted" memory until a restart is scheduled
+;opcache.max_wasted_percentage=5
+
+; When this directive is enabled, the OPcache appends the current working
+; directory to the script key, thus eliminating possible collisions between
+; files with the same name (basename). Disabling the directive improves
+; performance, but may break existing applications.
+;opcache.use_cwd=1
+
+; When disabled, you must reset the OPcache manually or restart the
+; webserver for changes to the filesystem to take effect
+opcache.validate_timestamps=1
+
+; How often (in seconds) to check file timestamps for changes to the shared
+; memory storage allocation. ("1" means validate once per second, but only
+; once per request. "0" means always validate)
+opcache.revalidate_freq=30
+
+; Enables or disables file search in include_path optimization
+;opcache.revalidate_path=0
+
+; If disabled, all PHPDoc comments are dropped from the code to reduce the
+; size of the optimized code.
+;opcache.save_comments=1
+
+; If disabled, PHPDoc comments are not loaded from SHM, so "Doc Comments"
+; may be always stored (save_comments=1), but not loaded by applications
+; that don't need them anyway.
+;opcache.load_comments=1
+
+; If enabled, a fast shutdown sequence is used for the accelerated code
+;opcache.fast_shutdown=0
+
+; Allow file existence override (file_exists, etc.) performance feature.
+;opcache.enable_file_override=0
+
+; A bitmask, where each bit enables or disables the appropriate OPcache
+; passes
+;opcache.optimization_level=0xffffffff
+
+;opcache.inherited_hack=1
+;opcache.dups_fix=0
+
+; The location of the OPcache blacklist file (wildcards allowed).
+; Each OPcache blacklist file is a text file that holds the names of files
+; that should not be accelerated. The file format is to add each filename
+; to a new line. The filename may be a full path or just a file prefix
+; (i.e., /var/www/x  blacklists all the files and directories in /var/www
+; that start with 'x'). Line starting with a ; are ignored (comments).
+;opcache.blacklist_filename=
+
+; Allows exclusion of large files from being cached. By default all files
+; are cached.
+;opcache.max_file_size=0
+
+; Check the cache checksum each N requests.
+; The default value of "0" means that the checks are disabled.
+;opcache.consistency_checks=0
+
+; How long to wait (in seconds) for a scheduled restart to begin if the cache
+; is not being accessed.
+; Default was 180 (3 minutes), changed to 1 hour to keep the
+; cache from being reset regularly.
+opcache.force_restart_timeout=3600
+
+; OPcache error_log file name. Empty string assumes "stderr".
+; opcache.error_log=/opt/meza/logs/opcache_error.log
+
+; All OPcache errors go to the Web server log.
+; By default, only fatal errors (level 0) or errors (level 1) are logged.
+; You can also enable warnings (level 2), info messages (level 3) or
+; debug messages (level 4).
+;opcache.log_verbosity_level=1
+
+; Preferred Shared Memory back-end. Leave empty and let the system decide.
+;opcache.preferred_memory_model=
+
+; Protect the shared memory from unexpected writing during script execution.
+; Useful for internal debugging only.
+;opcache.protect_memory=0

--- a/scripts/php.sh
+++ b/scripts/php.sh
@@ -7,12 +7,12 @@ print_title "Starting script php.sh"
 #
 # Prompt user for PHP version
 #
-while [ -z "$phpversion" ]
-do
-echo -e "\n\n\n\nVisit http://php.net/downloads.php"
-echo -e "\nEnter the version of PHP you would like (such as 5.4.42) and press [ENTER]: "
-read phpversion
-done
+# while [ -z "$phpversion" ]
+# do
+# echo -e "\n\n\n\nVisit http://php.net/downloads.php"
+# echo -e "\nEnter the version of PHP you would like (such as 5.4.42) and press [ENTER]: "
+# read phpversion
+# done
 
 
 #

